plugins/inputs/windows_event_log/wineventlog/wineventlog.go

```diff
@@ -104,32 +105,19 @@ func (l *windowsEventLog) Stop() {
 }
 
 func (l *windowsEventLog) run() {
-	recordNumber := l.eventOffset
 	ticker := time.NewTicker(1 * time.Second)
 	defer ticker.Stop()
 	for {
 		select {
 		case <-ticker.C:
-			records, err := l.read()
-			if err == RPC_S_INVALID_BOUND {
-				log.Printf("E! [windows_event_log] Due to corrupted/large event, skipping event log with record number %d, and log group name %s", recordNumber, l.logGroupName)
-				recordNumber = recordNumber + 1 // Advance to the next event to avoid being stuck
-				continue
-			}
-			if err != nil {
-				log.Printf("E! [windows_event_log] Failed to read Windows event logs for log group name %s. Details: %v\n", l.logGroupName, err)
-				recordNumber = recordNumber + 1
-				continue
-			}
-
```

Comment:
This code to track `recordNumber` is unecessary.
It gets parsed below and set in the `LogEvent` struct.
I moved printing the error inside the `read()` function.

Reply:
Let me know if I am crazy, but in the current `master` branch there is only 1 place where the `recordNumber` variable is used:
https://github.com/aws/amazon-cloudwatch-agent/blob/master/plugins/inputs/windows_event_log/wineventlog/wineventlog.go#L136

And just before it is used it is set:
https://github.com/aws/amazon-cloudwatch-agent/blob/master/plugins/inputs/windows_event_log/wineventlog/wineventlog.go#L131

There is no point in setting a local variable or incrementing it if it is not used.
I was confused by this code too :)

Reply:
`EvtNext()` is a stateful call, meaning each call will return a different  list of EvtHandles, or nothing.
Since the `windowsEventLog.read()` method calls `EvtNext()` it is also stateful.

MSDN:
https://docs.microsoft.com/en-us/windows/win32/wes/querying-for-events#reading-events-from-the-result-set
```
he events in the result set are not static; new events that are written to the channel will be included in the result set until ERROR_NO_MORE_ITEMS is set.
```

Reply:
Ah I see.  This almost seems like a bug then, if the following happened:

1. Read corrupted/large event from event log, get RPC_S_INVALID_BOUND error.  No more events available in the log yet.
2. Agent stops.  No updated offset is persisted in the state file.
3. Agent starts again, starts reading at the bad event again.
4. Repeat.

I guess that's an edge case though and would be resolved as soon as a new valid event is available in the log.